lib/cWarnings.ml

```diff
@@ -24,8 +24,12 @@ exception WarnError of w
 
 module DMap = PolyMap.Make (struct type nonrec 'a tag = 'a tag = .. end)
 module PrintMap = DMap.Map(struct type 'a t = 'a -> Pp.t end)
+module RevMap = DMap.Map(struct type 'a t = string end)
```

Comment:
This additional bookkeeping seems redundant with the Dyn interface to me. Isn't there a way to use it instead? (Also the tag values from this module are not safe to marshal, while the Dyn ones are.)

Reply:
NB this `RevMap` is not present in the new #17585